spec/api-subframe-spec.js

```diff
@@ -0,0 +1,88 @@
+const { expect } = require('chai')
+const { remote } = require('electron')
+const path = require('path')
+
+const { emittedNTimes, emittedOnce } = require('./events-helpers')
+const { closeWindow } = require('./window-helpers')
+
+const { BrowserWindow } = remote
+
+describe('renderer nodeSupportInSubFrames', () => {
+  const generateTests = (sandboxEnabled) => {
+    describe(`with sandbox ${sandboxEnabled ? 'enabled' : 'disabled'}`, () => {
+      let w
+
+      beforeEach(async () => {
+        await closeWindow(w)
+        w = new BrowserWindow({
+          show: false,
+          width: 400,
+          height: 400,
+          webPreferences: {
+            sandbox: sandboxEnabled,
+            preload: path.resolve(__dirname, 'fixtures/sub-frames/preload.js'),
+            nodeSupportInSubFrames: true
+          }
+        })
+      })
+
+      afterEach(() => {
+        return closeWindow(w).then(() => { w = null })
+      })
+
+      it('should load preload scripts in top level iframes', async () => {
+        const detailsPromise = emittedNTimes(remote.ipcMain, 'preload-ran', 2)
+        w.loadFile(path.resolve(__dirname, 'fixtures/sub-frames/frame-container.html'))
+        const [event1, event2] = await detailsPromise
+        expect(event1[0].frameId).to.not.equal(event2[0].frameId)
+        expect(event1[0].frameId).to.equal(event1[2])
+        expect(event2[0].frameId).to.equal(event2[2])
+      })
+
+      it('should load preload scripts in nested iframes', async () => {
+        const detailsPromise = emittedNTimes(remote.ipcMain, 'preload-ran', 3)
+        w.loadFile(path.resolve(__dirname, 'fixtures/sub-frames/frame-with-frame-container.html'))
+        const [event1, event2, event3] = await detailsPromise
```

Comment:
Would it make sense to refactor these tests to fail rather than timeout if the preload doesn't get loaded?

Reply:
Yes, but that's hard to do without complicated IPC.  A timeout in this case seemed the simplest (unfortunately) unless you have a good idea here